core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -152,168 +164,33 @@ private void cleanExpiredSnapshots() {
     // 2. Delete any data files that were deleted by those snapshots and are not in the table
     // 3. Delete any manifests that are no longer used by current snapshots
     // 4. Delete the manifest lists
+    SnapshotExpirationChanges snapshotExpirationChanges =
+        ExpireSnapshotUtil.getExpiredSnapshots(ops, base);
 
-    TableMetadata current = ops.refresh();
-
-    Set<Long> validIds = Sets.newHashSet();
-    for (Snapshot snapshot : current.snapshots()) {
-      validIds.add(snapshot.snapshotId());
-    }
-
-    Set<Long> expiredIds = Sets.newHashSet();
-    for (Snapshot snapshot : base.snapshots()) {
-      long snapshotId = snapshot.snapshotId();
-      if (!validIds.contains(snapshotId)) {
-        // the snapshot was expired
-        LOG.info("Expired snapshot: {}", snapshot);
-        expiredIds.add(snapshotId);
-      }
-    }
-
-    if (expiredIds.isEmpty()) {
+    if (snapshotExpirationChanges.getExpiredSnapshotIds().isEmpty()) {
       // if no snapshots were expired, skip cleanup
       return;
     }
 
-    LOG.info("Committed snapshot changes; cleaning up expired manifests and data files.");
-
-    cleanExpiredFiles(current.snapshots(), validIds, expiredIds);
+    LOG.info("Cleaning up expired manifests and data files locally.");
+    cleanExpiredFiles(
+        snapshotExpirationChanges.getCurrentSnapshots(),
+        snapshotExpirationChanges.getValidSnapshotIds(),
+        snapshotExpirationChanges.getExpiredSnapshotIds());
   }
 
-  @SuppressWarnings("checkstyle:CyclomaticComplexity")
-  private void cleanExpiredFiles(List<Snapshot> snapshots, Set<Long> validIds, Set<Long> expiredIds) {
+  private void cleanExpiredFiles(
+      List<Snapshot> currentTableSnapshots, Set<Long> validIds, Set<Long> expiredIds) {
```

Comment:
I think it makes sense to prepend `current` to the name to make it more explicit, but I'd remove `Table` from the name as it is obvious and makes the line longer.  If we rename it to `currentSnapshots`, args will fit on one line too.